### ISSUE TITLE
Paralle Tiled Inference

### ISSUE THREAD

**Issue by Joao-L-S-Almeida** [IBM/terratorch#370]
Tiled inference should support parallel evaluation for multiple workers

**Comment by romeokienzler** [comment]
@romeokienzler good idea - when talking about workers, u mean the torch.jit.fork or torch.distributed.launch package? Oder something else? We of course can implement parallel computation completely outside pytorch (with the advantage of hybrid and non-gang scheduled computing)

**Comment by Joao-L-S-Almeida** [comment]
Hi, @romeokienzler It could be for both the cases. We could distribute the processes under shared or distributed memory. I remember the initial goal was to improve the inference for single CPU inference, so I think it would be interesting to explore the shared memory parallelism.

**Comment by romeokienzler** [comment]
OK, so this is a super - issue as we can improve on various dimensions, we can reduce inference resource consumption (in some ways) but also introduce parallelism

I'm currently looking into if we can just throw resources at the problem and run inference on DDP

**Comment by Joao-L-S-Almeida** [comment]
Yes, if we are albe of using N GPUs (N > 1) we could simply use: https://pytorch.org/docs/stable/generated/torch.nn.DataParallel.html#dataparallel

**Comment by romeokienzler** [comment]
@Joao-L-S-Almeida just ran some tests with 1,2,4 GPUs on the same node and as expected the lighning trainer picks up all 4 GPU, creates 4 ranks and multiplexes the batches over the 4 GPUs, so I think for single node multiple GPUs we are good for now, need to work on the multi node multi GPU now
